nabo/nabo.cpp

```diff
@@ -87,36 +106,36 @@ namespace Nabo
 		if (allowSelfMatch)
 		{
 			if (k > cloud.cols())
-				throw runtime_error((boost::format("Requesting more points (%1%) than available in cloud (%2%)") % k % cloud.cols()).str());
+				throw runtime_error() << "Requesting more points (" << k << ") than available in cloud (" << cloud.cols() << ")";
 		}
 		else
 		{
 			if (k > cloud.cols()-1)
-				throw runtime_error((boost::format("Requesting more points (%1%) than available in cloud minus 1 (%2%) (as self match is forbidden)") % k % (cloud.cols()-1)).str());
+				throw runtime_error() << "Requesting more points (" << k << ") than available in cloud minus 1 (" << cloud.cols()-1 << ") (as self match is forbidden)";
 		}
 		if (query.rows() < dim)
-			throw runtime_error((boost::format("Query has less dimensions (%1%) than requested for cloud (%2%)") % query.rows() % dim).str());
+			throw runtime_error() << "Query has less dimensions (" << query.rows() << ") than requested for cloud (" << dim << ")";
 		if (indices.rows() != k)
-			throw runtime_error((boost::format("Index matrix has a different number of rows (%1%) than k (%2%)") % indices.rows() % k).str());
+			throw runtime_error() << "Index matrix has a different number of rows (" << indices.rows() << ") than k (" << k << ")";
 		if (indices.cols() != query.cols())
-			throw runtime_error((boost::format("Index matrix has a different number of columns (%1%) than query (%2%)") % indices.rows() % query.cols()).str());
+			throw runtime_error() << "Index matrix has a different number of columns (" << indices.rows() << ") than query (" << query.cols() << ")";
 		if (dists2.rows() != k)
-			throw runtime_error((boost::format("Distance matrix has a different number of rows (%1%) than k (%2%)") % dists2.rows() % k).str());
+			throw runtime_error() << "Distance matrix has a different number of rows (" << dists2.rows() << ") than k (" << k << ")";
 		if (dists2.cols() != query.cols())
-			throw runtime_error((boost::format("Distance matrix has a different number of columns (%1%) than query (%2%)") % dists2.rows() % query.cols()).str());
+			throw runtime_error() << "Distance matrix has a different number of columns (" << dists2.rows() << ") than query (" << query.cols() << ")";
 		if (maxRadii && (maxRadii->size() != query.cols()))
-			throw runtime_error((boost::format("Maximum radii vector has not the same length (%1%) than query has columns (%2%)") % maxRadii->size() % k).str());
+			throw runtime_error() << "Maximum radii vector has not the same length (" << maxRadii->size() << ") than query has columns (" << k << ")";
 		const unsigned maxOptionFlagsValue(ALLOW_SELF_MATCH|SORT_RESULTS);
 		if (optionFlags > maxOptionFlagsValue)
-			throw runtime_error((boost::format("OR-ed value of option flags (%1%) is larger than maximal valid value (%2%)") % optionFlags % maxOptionFlagsValue).str());
+			throw runtime_error() << "OR-ed value of option flags (" << optionFlags << ") is larger than maximal valid value (" << maxOptionFlagsValue << ")";
 	}
-	
```

Comment:
Personally I suggest to remove these whitespace changes from the patch. The reason these whitespaces are here is that in my original coding conventions (never written down, my bad), I did not trim whitespaces on empty lines, as I think that they make the structure more readable in an editor that has a light indication for tabs. @simonlynen, are there coding conventions now regarding that point?